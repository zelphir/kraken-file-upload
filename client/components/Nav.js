import React from 'react'
import PropTypes from 'prop-types'
import { Navbar, NavbarGroup, NavbarDivider, NavbarHeading } from '@blueprintjs/core'
import UploadButton from './UploadButton'

const Nav = ({ hasData, onUpload }) => {
  return (
    <Navbar className="pt-dark" fixedToTop>
      <NavbarGroup>
        <NavbarHeading>Kraken test</NavbarHeading>
        {hasData && (
          <React.Fragment>
            <NavbarDivider />
            <UploadButton onUpload={onUpload} />
          </React.Fragment>
        )}
      </NavbarGroup>
      <NavbarGroup align="right">
        <div className="pt-input-group">
          <span className="pt-icon pt-icon-search" />
          <input
            className="pt-input pt-fill"
            style={{ minWidth: 240 }}
            type="search"
            placeholder="Search by name or type"
            dir="auto"
          />
        </div>
      </NavbarGroup>
    </Navbar>
  )
}

Nav.propTypes = {
  hasData: PropTypes.bool.isRequired,
  onUpload: PropTypes.func.isRequired
}

export default Nav
