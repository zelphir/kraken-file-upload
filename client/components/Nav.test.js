import React from 'react'
import { shallow } from 'enzyme'
import Nav from './Nav'

function setup(nextProps = {}) {
  const props = {
    hasData: false,
    hasQuery: false,
    onUpload: jest.fn(),
    handleFilter: jest.fn(),
    ...nextProps
  }
  const wrapper = shallow(<Nav {...props} />)

  return {
    props,
    wrapper
  }
}

describe('<Nav />', () => {
  it('should match the snapshot', () => {
    const { wrapper } = setup()
    expect(wrapper).toMatchSnapshot()
  })

  it('should render the divider and Add file button if has data', () => {
    const { wrapper } = setup({ hasData: true })
    expect(wrapper.find('UploadButton')).toHaveLength(1)
    expect(wrapper.find('NavbarDivider')).toHaveLength(1)
  })
})
