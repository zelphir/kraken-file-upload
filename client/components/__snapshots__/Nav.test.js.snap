// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> should match the snapshot 1`] = `
<Blueprint2.Navbar
  className="pt-dark"
  fixedToTop={true}
>
  <Blueprint2.NavbarGroup
    align="left"
  >
    <Blueprint2.NavbarHeading>
      Kraken test
    </Blueprint2.NavbarHeading>
  </Blueprint2.NavbarGroup>
  <Blueprint2.NavbarGroup
    align="right"
  >
    <div
      className="pt-input-group"
    >
      <span
        className="pt-icon pt-icon-search"
      />
      <input
        className="pt-input pt-fill"
        dir="auto"
        disabled={true}
        onChange={[MockFunction]}
        placeholder="Search by name or type"
        style={
          Object {
            "minWidth": 240,
          }
        }
        type="search"
      />
    </div>
  </Blueprint2.NavbarGroup>
</Blueprint2.Navbar>
`;
